daisy/instance.go

```diff
@@ -371,13 +372,25 @@ func newInstanceRegistry(w *Workflow) *instanceRegistry {
 	return ir
 }
 
+// Sleep function is mocked on testing.
+var SleepFn = time.Sleep
+
 func (ir *instanceRegistry) deleteFn(res *Resource) dErr {
 	m := namedSubexp(instanceURLRgx, res.link)
-	err := ir.w.ComputeClient.DeleteInstance(m["project"], m["zone"], m["instance"])
-	if gErr, ok := err.(*googleapi.Error); ok && gErr.Code == http.StatusNotFound {
-		return typedErr(resourceDNEError, err)
+	for {
```

Comment:
If the instance is deleted outside of the workflow then this will just spin forever, Is this because of a race condition between a create step, a failure, and a cleanup?

Reply:
> If the instance is deleted outside of the workflow then this will just spin forever

A daisy instance removed by someone else or on a different daisy step? Is that a concern?

> Is this because of a race condition between a create step, a failure, and a cleanup?

Yes. The creation step takes a while, and while it's creating, the CopyGCSObject fails and triggers the cleanup. The cleanup runs this instance deletion code but despite finding information about the instance in the internal registry, the googleapi will not find the instance for removing (the creation didn't finish, the instance is not even on PROVISIONING state!). That was why I implemented this wait mechanism.

Reply:
Deleting resources is definitely a thing that can happen, I would be ok with just a limit on retries here for now, maybe later we can come up with a better state management system for active compute calls.